### ISSUE TITLE
Add delayed job processing queue to Rails for background job processing

### ISSUE THREAD

**Issue by r-bartlett-gsa** [GSA/Challenge_platform#42]
### User Story
As a user, I want the server to be responsive and provide immediate feedback about tasks that take more than a few milliseconds.

**Note**: this work will be required to migrate the Exports from Phoenix to Rails.

**Acceptance criteria:**

- [ ] When this task is complete, Rails developers will be able to queue up delayed jobs to free up system resources for faster response times.
- [ ] Emails are queued in the background to improve response times from the server.
- [ ] Exports jobs can be queued from Rails.

### Definition of Done

**Doing (dev team)**
- [ ] Code complete 
- [ ] Code is organized appropriately
- [ ] Any known trade offs are documented in the associated GH issue
- [ ] Code is documented, modules, shared functions, etc.
- [ ] Automated testing has been added or updated in response to changes in this PR
- [ ] The feature is smoke tested to confirm it meets requirements
- [ ] Database changes have been peer reviewed for index changes and performance bottlenecks
- [ ] PR that changes or adds UI
  - [ ] include a screenshot of the WAVE report for the altered pages
  - [ ] Confirm changes were validated for mobile responsiveness
- [ ] PR approved / Peer reviewed 
- [ ] Move card to testing column in the board
---
**Testing (dev team)**
- [ ] Security scans passed
- [ ] Automate accessibility tests passed
- [ ] Build process and deployment is automated and repeatable
- [ ] Feature toggles if appropriate
- [ ] Deploy to staging

**Staging**
- [ ] Accessibility tested (Marni)
  - [ ]  Keyboard navigation
  - [ ]  Focus confirmed
  - [ ] Color contrast compliance
  - [ ] Screen reader testing 
- [ ] Usability testing: mobile and desktop (Tracy or Marni)
- [ ] Cross browser testing (tool to be determined) (Tracy or Marni)
  - [ ]  UI rendering is performant
- [ ] AC review (Renata)
- [ ] Deploy to production (production-like environment for eval capability) (dev team)
- [ ] Move to production column in the board
---
**Production**
- [ ] User and security documentation has been reviewed for necessary updates (Renata and Michelle) 
- [ ] PO / PM approved (Jarah or Renata)
- [ ] AC is met and it works as expected (Jarah or Renata)
- [ ] Move to done column in the board (Jarah or Renata)

**Comment by danivovich** [comment]
@r-bartlett-gsa I don't think this is actually turning into a requirement for the MVP work, or even the MMP work. Since communication is transactional it can be done within the web request, and the background processing for other things was reused from the portal for download and similar tasks.

**Comment by r-bartlett-gsa** [comment]
> [@r-bartlett-gsa](https://github.com/r-bartlett-gsa) I don't think this is actually turning into a requirement for the MVP work, or even the MMP work. Since communication is transactional it can be done within the web request, and the background processing for other things was reused from the portal for download and similar tasks.

@danivovich Sounds good. Thank you for letting me know.

**Comment by afeijoo** [comment]
Archiving repo; no longer applicable.
